webapp/src/main/java/com/box/l10n/mojito/service/assetExtraction/AssetExtractionService.java

```diff
@@ -444,6 +454,11 @@ CreateTextUnitsResult createTextUnitsForNewContent(
           ImmutableList<BranchStateTextUnit> toCreateTmTextUnits =
               getBranchStateTextUnitsWithoutId(stateForNewContentWithIds);
 
+          final Repository repository = assetContent.getBranch().getRepository();
+          if (toCreateTmTextUnits.isEmpty() && repository != null) {
```

Comment:
Is the intention here that if `toCreateTmTextUnits` is empty that means that the branch will not show in the branch page as no text units are created so no stats job is triggered?

Can you put a comment in the code explaining this if that's the case